lms/djangoapps/certificates/models.py

```diff
@@ -878,20 +889,56 @@ def is_enabled_for_course(cls, course_key):
         except cls.DoesNotExist:
             return False
         else:
-            return latest.enabled
+            return latest.can_self_generate
 
     @classmethod
-    def set_enabled_for_course(cls, course_key, is_enabled):
+    def set_self_generation_for_course(cls, course_key, enable):
         """Enable or disable self-generated certificates for a course.
 
         Arguments:
             course_key (CourseKey): The identifier for the course.
-            is_enabled (boolean): Whether to enable or disable self-generated certificates.
+            enable (boolean): Whether to enable or disable self-generated certificates.
+
+        """
+        defaults = {
+            'enabled': enable,
+            'can_self_generate': enable
+        }
+        CertificateGenerationCourseSetting.objects.update_or_create(
+            course_key=course_key,
+            defaults=defaults
+        )
+
+    @classmethod
+    def is_language_specific_templates_enabled_for_course(cls, course_key):
```

Comment:
Nit: I think it would be clearer without the the `is_` prefix.

Reply:
Nit: If you end up changing this field name to can_use_language_specific_templates, it would also be nice to change this to `course_can_use_language_specific_templates`.